aten/src/ATen/Device.h

```diff
@@ -112,3 +113,16 @@ struct Device {
 
 std::ostream& operator<<(std::ostream& stream, at::Device::Type type);
 std::ostream& operator<<(std::ostream& stream, const at::Device& device);
+
+namespace std {
+  template<> struct hash<at::Device>
+  {
+    int64_t operator()(const at::Device& device) const noexcept {
```

Comment:
If we wanted to be pedantic, the return value of `std::hash::operator()` has to be `size_t`: https://en.cppreference.com/w/cpp/utility/hash/operator()